docs/components/properties.md

```diff
@@ -567,6 +567,61 @@ and TSX:
 <todo-list-item isComplete={false}></todo-list-item>
 ```
 
+When using union types, the type of a component's `@Prop()` value can be ambiguous at runtime.
+In the provided example, under what circumstances does `@Prop() isComplete` function as a `string`, and when does it serve as a `boolean`?
+
+When using a component in HTML, the runtime value of a `@Prop()` is always a string.
+This is a result of setting the HTML attribute for the custom element.
```

Comment:
Should we possibly mention that this is in-line with how attributes work on html elements in general? I don't know, might be an opportunity to do a little web fundamentals education for the reader which is helpful for understanding the 'why' here

Reply:
I go back and forth here - on one hand, I think it'd be beneficial to give folks some background knowledge that are coming to Stencil/WC in general. On the other, I don't think it's our responsibility to teach HTML to folks. Back on the original hand, there's some trickiness with Stencil/TS and marrying some of those types up with HTML types.

I made some copy edits in aa91e1e39 - LMK what you think. I held off on mentioning the spec/talking about "standards" too much here, although I think there's some room for improvement.

Maybe this aligns well with some of the thoughts/things @tanner-reits was talking about during our last JEDI days for reorganizing the docs a bit - specifically I wonder if we should revisit our onboarding experience to consider some WC fundamentals needed to use Stencil effectively 🤔 

Since don't align with the spec in some cases, and I can see a world where saying "Here's the spec, here's how to work effectively using it, and here are the special things Stencil does" having some value for users. Thoughts? (Out of scope for this PR obviously 🙃)

Reply:
Sure, that edit looks good to me - my thought on mentioning this is part of attributes in general is just providing a bit of the 'why' for this dichotomy in behavior b/tw attrs specified in HTML vs using `setAttribute`, I've always found that if you have a 'why' explanation it's easier to remember something like this as opposed to if it's just presented as sort of an arbitrary 'this is how it is.' But I think the edit is good as is.

On the larger rewrite I think that's a good idea, in general it would be good I think for developers to have a good intuition about where Stencil ends and custom elements begin, I don't really know how much that would be understood by the average Stencil user at present but if our docs could be structured to facilitate that understanding I think that would be a step forward